File: app/src/main/java/com/sample/wipro/MainActivity.java
package com.sample.wipro;

import android.support.v4.widget.SwipeRefreshLayout;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;

import java.util.ArrayList;
import java.util.Arrays;
import java.util.Collections;
import java.util.Random;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        SwipeRefreshLayout swipeRefreshLayout;
        RecyclerView recyclerView;
        // ArrayList for person names
        ArrayList personNames = new ArrayList<>(Arrays.asList("Person 1", "Person 2", "Person 3", "Person 4", "Person 5", "Person 6", "Person 7", "Person 8", "Person 9", "Person 10", "Person 11", "Person 12", "Person 13", "Person 14"));
        ArrayList personImages = new ArrayList<>(Arrays.asList(R.drawable.person1, R.drawable.person2, R.drawable.person3, R.drawable.person4, R.drawable.person5, R.drawable.person6, R.drawable.person7, R.drawable.person1, R.drawable.person2, R.drawable.person3, R.drawable.person4, R.drawable.person5, R.drawable.person6, R.drawable.person7));

        @Override
        protected void onCreate(Bundle savedInstanceState) {
            super.onCreate(savedInstanceState);
            setContentView(R.layout.activity_main);
            // init SwipeRefreshLayout and ListView
            swipeRefreshLayout = (SwipeRefreshLayout) findViewById(R.id.simpleSwipeRefreshLayout);
            // get the reference of RecyclerView
            recyclerView = (RecyclerView) findViewById(R.id.recyclerView);
            // set a LinearLayoutManager with default vertical orientation
            LinearLayoutManager linearLayoutManager = new LinearLayoutManager(getApplicationContext());
            recyclerView.setLayoutManager(linearLayoutManager);
            // call the constructor of CustomAdapter to send the reference and data to Adapter
            CustomAdapter customAdapter = new CustomAdapter(MainActivity.this, personNames, personImages);
            recyclerView.setAdapter(customAdapter); // set the Adapter to RecyclerView
            // implement setOnRefreshListener event on SwipeRefreshLayout
            swipeRefreshLayout.setOnRefreshListener(new SwipeRefreshLayout.OnRefreshListener() {
                @Override
                public void onRefresh() {
                    // cancel the Visual indication of a refresh
                    swipeRefreshLayout.setRefreshing(false);
                    shuffleItems();
                }
            });
        }

        public void shuffleItems() {
            // shuffle the ArrayList's items and set the adapter
            Collections.shuffle(personNames, new Random(System.currentTimeMillis()));
            Collections.shuffle(personImages, new Random(System.currentTimeMillis()));
            // call the constructor of CustomAdapter to send the reference and data to Adapter
            CustomAdapter customAdapter = new CustomAdapter(MainActivity.this, personNames, personImages);
            recyclerView.setAdapter(customAdapter); // set the Adapter to RecyclerView
        }
    }
}

File: app/src/main/java/com/sample/wipro/MainActivity.kt
package com.sample.wipro

import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import android.support.v7.widget.LinearLayoutManager
import android.support.v7.widget.RecyclerView
import android.support.v7.widget.Toolbar
import android.widget.LinearLayout
import com.android.volley.Request
import com.android.volley.Response
import com.android.volley.toolbox.StringRequest
import com.android.volley.toolbox.Volley
import org.json.JSONArray
import org.json.JSONObject

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        var items  = ArrayList<MyData>()

        val _recyclerView: RecyclerView = findViewById(R.id.recycler)

        _recyclerView.layoutManager = LinearLayoutManager(this, LinearLayout.VERTICAL, false)

        val queue = Volley.newRequestQueue(this)

        val url: String = "https://dl.dropboxusercontent.com/s/2iodh4vg0eortkl/facts.json"

        // Request a string response from the provided URL.
        val stringReq = StringRequest(Request.Method.GET, url,
                Response.Listener<String> { response ->

                    var strResp = response.toString()
                    val jsonObj: JSONObject = JSONObject(strResp)
                    var heading:String = jsonObj.get("title").toString()
                    setTitle(heading)

                    val jsonArray: JSONArray = jsonObj.getJSONArray("rows")

                    for (i in 0 until jsonArray.length()) {
                        var jsonInner: JSONObject = jsonArray.getJSONObject(i)
                        //adding some dummy data to the list

                        items.add(MyData(jsonInner.get("title").toString(),jsonInner.get("description").toString(),jsonInner.get("imageHref").toString()))
                    }
                    val adapter = RecyclerViewAdapter(items)
                    _recyclerView.adapter = adapter
                },
                Response.ErrorListener {  })
        queue.add(stringReq)
    }

}





